diff --git a/Template/LO/SubProcesses/genps.f b/Template/LO/SubProcesses/genps.f
index 258cbd147..e2a5bc1c2 100644
--- a/Template/LO/SubProcesses/genps.f
+++ b/Template/LO/SubProcesses/genps.f
@@ -1591,11 +1591,34 @@ C     -----------------------------------------
 
       integer ids(nexternal)
       integer i,j
+      logical trivial_boost 
 
 c     uncompress
       call mapid(frame_id, ids)
       pboost(:) = 0d0
       p2(:,:) = 0d0
+c     avoid trivial boost (check only for  0 0 1 1 1... so sum of the
+c     final state. 1 1 0 0 0 .... should not go within this function   
+      trivial_boost=.true.
+      do i=1,nexternal
+        if (i.le.nincoming)then
+            if (ids(i).ne.0)then
+                trivial_boost=.false.
+                EXIT ! stop do loop
+            endif
+        else
+            if (ids(i).ne.1)then
+                  trivial_boost=.false.
+                  EXIT ! stop do loop
+              endif
+        endif
+        enddo
+
+        if (trivial_boost)then
+            p2(:,:) = p1(:,:)
+            return
+        endif
+
 c     find the boost momenta --sum of particles--
       do i=1,nexternal
        if (ids(i).eq.1)then
